### PR TITLE
Make rvs_to_values work with non-RandomVariables

### PR DESCRIPTION
**Pull request by ricardoV94**
**What is this PR about?**
Cleanup some logic that assumed only RandomVariables can be valued

Related to #6072 
Related to https://github.com/aesara-devs/aeppl/issues/174

**Checklist**
+ [x] Explain important implementation details 👆
+ [x] Make sure that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html).
+ [x] Link relevant issues (preferably in [nice commit messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html))
+ [x] Are the changes covered by tests and docstrings?
+ [x] Fill out the short summary sections 👇

## Major / Breaking Changes
- ...

## Bugfixes / New features
- rvs_to_values now works with non-RandomVariables and unvalued RandomVariables

## Docs / Maintenance
- ...

### CODE REVIEW

File: pymc/aesaraf.py [hunk 1]
```diff
@@ -198,22 +170,40 @@ def extract_obs_data(x: TensorVariable) -> np.ndarray:
     raise TypeError(f"Data cannot be extracted from {x}")
 
 
+def extract_rv_and_value_vars(
+    random_var: TensorVariable,
+) -> Tuple[TensorVariable, TensorVariable]:
```
Comment (lucianopaz) [inline pymc/aesaraf.py]: The type hint is wrong here. It should say that the `TensorVariable`s are `Optional`.

File: pymc/aesaraf.py [hunk 2]
```diff
@@ -198,22 +170,40 @@ def extract_obs_data(x: TensorVariable) -> np.ndarray:
...
+        A variable corresponding to a ``RandomVariable``.
+
+    Returns
+    =======
+    The first value in the tuple is the ``RandomVariable``, and the second is the
+    measure/log-likelihood value variable that corresponds with the latter.
+
+    """
+    if not random_var.owner:
+        return None, None
```
Comment (lucianopaz) [inline pymc/aesaraf.py]: It looks like this isn’t tested. Why should it return `None` instead of raising something like a `TypeError`?

**Review by lucianopaz — CHANGES_REQUESTED** [review]
Looks ok @ricardoV94 . I’m entirely sure about how the model graphs are walked and the replacements are being set. But I think it’s just that I’m tired and I’ll have to review it a second time. Anyway, I wanted to ask you to change a type hint that I think is wrong and maybe also explain why you don’t raise an exception instead of returning Nones

File: pymc/aesaraf.py (hunk 2 above)
Comment (ricardoV94) [reply pymc/aesaraf.py]: I think this is for input variables that don't have an `owner` like `at.scalar()`. I can however have the check outside. The whole function is probably not needed...

**Review by twiecki — APPROVED** [review]

**Pull request merged**